lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -48,7 +52,27 @@ DartType? getExpectedType(PostfixExpression node) {
       return null;
     }
     var staticType = parentExpression.staticType;
-    return staticType is FunctionType ? staticType.returnType : null;
+    if (staticType is! FunctionType) {
+      return null;
+    }
+    if (withAwait) {
+      //  return type is obligatorily a Future<X>
+      return (staticType.returnType as ParameterizedType?)?.typeArguments.first;
+    } else {
+      return staticType.returnType;
+    }
+  }
+  // in yield value
+  if (parent is YieldStatement) {
+    var parentExpression = parent.thisOrAncestorOfType<FunctionExpression>();
+    if (parentExpression == null) {
+      return null;
+    }
+    var staticType = parentExpression.staticType;
+    if (staticType is! FunctionType) {
+      return null;
+    }
+    return (staticType.returnType as ParameterizedType).typeArguments.first;
```

Comment:
Same here. Having a `yield` means that the return type is either `Iterable<X>` or `Stream<X>`. So a ParameterizedType.

Reply:
my bad, sorry and thanks